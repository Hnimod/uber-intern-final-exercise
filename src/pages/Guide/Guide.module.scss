.overlay {
  background-color: rgba(255, 255, 255, 0.9);
  max-width: 90rem;
  height: 100%;
  padding: 2rem;
  margin: 0 auto;
  flex: 1;
}

.content {
  margin-top: 3rem;
}

.introduction {
  margin-bottom: 5rem;
  line-height: 30px;
  font-weight: 600;
}

.steps {
  & > h3:not(:first-child) {
    margin-top: 2.5rem;
  }

  & > p {
    margin-top: 1rem;
    word-spacing: 3px;
    line-height: 30px;
    color: rgb(88, 88, 88);
  }
}

@media screen and (min-width: 768px) {
  .content {
    display: flex;
    justify-content: space-between;
  }

  .introduction {
    flex-basis: 40%;
    max-width: 30rem;
  }

  .steps {
    flex-basis: 50%;
  }
}
