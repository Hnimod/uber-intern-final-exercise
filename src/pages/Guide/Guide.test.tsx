import renderer from 'react-test-renderer';
import Guide from './Guide';

jest.mock('../../layouts/BackgroundImage', () => 'BackgroundImage');

it('should render commitment page correctly', () => {
  const tree = renderer.create(<Guide />).toJSON();
  expect(tree).toMatchSnapshot();
});
