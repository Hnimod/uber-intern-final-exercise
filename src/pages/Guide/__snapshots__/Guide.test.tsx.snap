// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render commitment page correctly 1`] = `
<BackgroundImage>
  <div
    className="overlay"
  >
    <h1
      className="title"
    >
      How Uber works
    </h1>
    <section
      className="content"
    >
      <p
        className="introduction"
      >
        On-demand transportation technology is our core service, and the app that connects drivers-partners and riders is what makes it all possible. Here's how it works, step by step:
      </p>
      <section
        className="steps"
      >
        <h3>
          Step 1: A rider opens the app
        </h3>
        <p>
          The rider enters their destination into the Where to? box on the top of the screen, taps each ride option to see the wait time, car sizes, and price; then confirms their pickup location and taps Request.
        </p>
        <h3>
          Step 2: The rider is matched with a driver
        </h3>
        <p>
          A nearby driver sees and chooses to accept the rider's trip request. the rider is automatically notified when the driver's vehicle is about a minute away
        </p>
        <h3>
          Step 3: The rider picks up the rider
        </h3>
        <p>
          The driver and the rider verify each other's names and the destination. Then the driver starts the ride.
        </p>
        <h3>
          Step 4: The driver takes the rider to the destination
        </h3>
        <p>
          The app gives the driver the option to access turn-by-turn directions. so the driver can focus on getting there and the rider can focus on enjoying a comfortable ride
        </p>
        <h3>
          Step 5: The driver and rider leave ratings and reviews
        </h3>
        <p>
          At the end of each trip, drivers and riders can rate each other from 1 to 5 stars. Riders can also give the driver compliments. In cities where tipping is available, they can also add a little extra to show their gratitude.
        </p>
      </section>
    </section>
  </div>
</BackgroundImage>
`;
